lib/db_utils.pm

```diff
@@ -179,3 +181,51 @@ sub push_image_data_to_db {
         record_soft_failure("poo#113120 - There has been a problem pushing data to the DB.");
     }
 }
+
+sub check_postgres_db {
+    my $image = shift;
+    my $db_ip = get_var('POSTGRES_IP');
+    my $db_port = get_var('POSTGRES_PORT', '5444');
+    my $db_log = "/tmp/db.log";
+    # Check if the running job can push data to the Data Base
+    # We only allow this on certain conditions:
+    #  - The job must contain IMAGE_MON_SERVER
+    #  - The job shouldn't be a verification run
+    #  - The job must be executed from OSD or O3
+    return 0 if (!$db_port && !$db_ip);
+
+    # CASEDIR var is always set, and if when no specified, the value is "sle" for OSD and "opensuse" for O3
+    #return 0 if (get_required_var('CASEDIR') !~ m/^sle$|^opensuse$|^(sle|leap)-micro$/);
+
+    my $job_url;
+    my $openqa_host = get_required_var('OPENQA_HOSTNAME');
+    if ($openqa_host =~ /openqa1-opensuse|openqa.opensuse.org/) {    # O3 hostname
+        $job_url = 'https://openqa.opensuse.org/tests/' . get_current_job_id();
+    }
+    elsif ($openqa_host =~ /openqa.suse.de/) {    # OSD hostname
+        $job_url = 'https://openqa.suse.de/tests/' . get_current_job_id();
+    }
+    return 0 unless ($job_url);
+
+    # Check if the data for this image has been already published before.
+    # This will avoid publishing the same data if a job is restarted.
+    # e.g.
+    # curl http://k2.qe.suse.de:8080/size\?product\=fts.jeos\&asset\=fts.SLES15-SP5-Minimal-VM.x86_64-kvm-and-xen-Build3.32.qcow2\&value\=eq.258.1875
+    my $query = sprintf('curl -I "http://%s:%s/size?product=fts.jeos&asset=fts.%s"', $db_ip, $db_port, $image);
+
+    script_run("echo '$query' | tee -a $db_log");
+    my $query_output = script_output("$query 2>&1 | tee -a $db_log", proceed_on_failure => 1);
+
+    # Successful query should return 'HTTP/1.1 200 OK'
+    # Empty records will return  '{"results":[{"statement_id":0}]}'
+    # Existing records will return '{"results":[{"statement_id":0,"series":[{"name":"size","columns":["time","value"],"values"' ...
+    if ($query_output !~ /200 OK/) {
+        record_soft_failure("poo#110221 - There has been a problem with the query.");
+        return 0;
+    } elsif ($query_output =~ /values/) {
+        record_info('DB skip', "Skip pushing the data to the DB. There is already a DB record for this image.");
```

Comment:
Shouldn't this SKIP be part of the push method? We are not pushing anything here, right?